src/main/java/Task.java

```diff
@@ -0,0 +1,68 @@
+public class Task{
+    protected static int taskCount = 0;
+    protected String description;
+    protected boolean isDone;
+
+    //constructor
+    public Task(String description){
+        this.description = description;
+        this.isDone = false;
+        taskCount += 1;
+    }
+    public String getStatus(){
+        return (isDone ? "X": " ");
+    }
+
+    public static int getTaskCount() {
+        return taskCount;
+    }
+
+    public void markTask(){
+        this.isDone = true;
+    }
+    public void unmarkTask(){
+        this.isDone = false;
+    }
+    @Override
+    public String toString(){
+        return "[" + getStatus() + "] " + this.description;
+    }
+}
+
+//Inheritance and Polymorphism
+class Deadline extends Task{
+
+    protected String by;
+    public Deadline(String description, String by){
+        super(description);
+        this.by = by;
+    }
+    @Override
+    public String toString() {
+        return "[D]"+ super.toString() + " (by: " + by + ")";
```

Comment:
Perhaps a space before +? I noticed the same issue in several other places too.
```suggestion
        return "[D]" + super.toString() + " (by: " + by + ")";
```